devicetypes/smartthings/zigbee-metering-dimmer.src/zigbee-metering-dimmer.groovy

```diff
@@ -0,0 +1,133 @@
+/**
+ *  Copyright 2021 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+metadata {
+	definition (name: "ZigBee Metering Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.light",  mnmn: "SmartThings", vid:"generic-dimmer-power-energy") {
+
+		capability "Actuator"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Switch"
+		capability "Switch Level"
+		capability "Health Check"
+
+		// Enbrighten/Jasco
+		fingerprint manufacturer: "Jasco Products", model: "43082", deviceJoinName: "Enbrighten Switch" //Enbrighten, in-Wall Smart Dimmer With Energy Monitoring 43082, Raw Description: 01 0104 0101 00 08 0000 0003 0004 0005 0006 0008 0702 0B05 02 000A 0019
+	}
+}
+
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+	List result = []
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.info event
+		if (event.name== "power") {
+			event.value = event.value/getPowerDiv()
+			event.unit = "W"
```

Comment:
does `getEvent` not set these units for you?

Reply:
I don't think so, this is a snippet from the live logging (lines 42 and 50 in the DTH)
923ea31c-a6a6-45c6-8487-5344368df472 2:47:03 PM: info event outer:[name:power, value:22.1, unit:W]
923ea31c-a6a6-45c6-8487-5344368df472 2:47:03 PM: info [name:power, value:221]